src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -216,6 +206,46 @@ public synchronized RcaResponse readRca(String rca) {
     return response;
   }
 
+  private RcaResponse readTemperatureProfileRca(String rca) {
+    RcaResponse response = null;
+    Field<Integer> primaryKeyField = DSL.field(
+        SQLiteQueryUtils.getPrimaryKeyColumnName(ResourceFlowUnit.RCA_TABLE_NAME), Integer.class);
+    SelectJoinStep<Record> rcaQuery = SQLiteQueryUtils.buildRcaQuery(create, rca);
+    try {
+      List<Record> recordList = rcaQuery.fetch();
+      if (recordList == null || recordList.isEmpty()) {
+        return null;
+      }
+      Record mostRecentRecord = recordList.get(0);
+      response = RcaResponse.buildResponse(mostRecentRecord);
+
+      if (rca.equals(ClusterTemperatureRca.TABLE_NAME)) {
+        Field<Integer> foreignKeyField = DSL.field(
+            SQLiteQueryUtils.getPrimaryKeyColumnName(ResourceFlowUnit.RCA_TABLE_NAME),
+            Integer.class);
+        SelectJoinStep<Record> query = SQLiteQueryUtils
+            .buildSummaryQuery(create, ClusterTemperatureSummary.TABLE_NAME,
+                mostRecentRecord.get(primaryKeyField),
+                foreignKeyField);
+        Result<Record> temperatureSummary = query.fetch();
+        GenericSummary summary =
+            ClusterTemperatureSummary.buildSummaryFromDatabase(temperatureSummary, create);
+        response.addNestedSummaryList(summary);
+      } else if (rca.equalsIgnoreCase(NodeTemperatureRca.TABLE_NAME)) {
```

Comment:
Should we also check for the node Role and ensure it is a data node we are making a query to ?

Reply:
With this change, I have added a 
I think we'll need to tweak the way we write/read node roles today.
For a data-master node, we only get either data or master.

Reply:
awesome ! thanks